Setup/Patch/Data/UpdateMerchantIdPath.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+namespace Alma\MonthlyPayments\Setup\Patch\Data;
+
+use Alma\API\Entities\Merchant;
+use Alma\API\RequestError;
+use Alma\MonthlyPayments\Gateway\Config\Config;
+use Alma\MonthlyPayments\Helpers\AlmaClient;
+use Alma\MonthlyPayments\Helpers\ConfigHelper;
+use Alma\MonthlyPayments\Helpers\Exceptions\AlmaClientException;
+use Alma\MonthlyPayments\Helpers\Logger;
+use Magento\Framework\App\Config\ScopeConfigInterface;
+use Magento\Framework\App\ResourceConnection;
+use Magento\Framework\Setup\Patch\DataPatchInterface;
+use Magento\Payment\Gateway\Config\Config as ConfigGateway;
+
+class UpdateMerchantIdPath implements DataPatchInterface
+{
+    const PATH_EQUAL = 'path = ?';
+    /**
+     * @var Logger
+     */
+    private $logger;
+    /**
+     * @var ResourceConnection
+     */
+    private $resourceConnection;
+    /**
+     * @var AlmaClient
+     */
+    private $almaClient;
+    /**
+     * @var ScopeConfigInterface
+     */
+    private $scopeConfig;
+    /**
+     * @var ConfigHelper
+     */
+    private $configHelper;
+
+    public function __construct(
+        Logger $logger,
+        ResourceConnection $resourceConnection,
+        ScopeConfigInterface $scopeConfig,
+        ConfigHelper $configHelper,
+        AlmaClient $almaClient
+    ) {
+        $this->logger = $logger;
+        $this->resourceConnection = $resourceConnection;
+        $this->almaClient = $almaClient;
+        $this->scopeConfig = $scopeConfig;
+        $this->configHelper = $configHelper;
+    }
+
+    /**
+     * @return array
+     */
+    public static function getDependencies(): array
+    {
+        return [];
+    }
+
+    /**
+     * @return array
+     */
+    public function getAliases(): array
+    {
+        return [];
+    }
+
+    /**
+     * @return void
+     * @throws RequestError
+     */
+    public function apply(): void
+    {
+        $oldMerchantIdPath = sprintf(ConfigGateway::DEFAULT_PATH_PATTERN, Config::CODE, 'merchant_id');
+
+        $tableName = $this->getCoreTable();
+        $oldMerchantIdQuery = $this->resourceConnection->getConnection()->select()
+            ->from($tableName, ['value'])->where(self::PATH_EQUAL, $oldMerchantIdPath);
+
+        $oldMerchant = new Merchant(['id' => $this->resourceConnection->getConnection()->fetchOne($oldMerchantIdQuery)]);
+
+        $apiQuery = $this->resourceConnection->getConnection()->select()
+            ->from($tableName, ['config_id','scope','scope_id', 'path', 'value'])
+            ->where(self::PATH_EQUAL, 'payment/alma_monthly_payments/live_api_key')
+            ->orWhere(self::PATH_EQUAL, 'payment/alma_monthly_payments/test_api_key');
+
+        $apiKeysRows = $this->resourceConnection->getConnection()->fetchAll($apiQuery);
+
+        foreach ($apiKeysRows as $apiKeysRow) {
+            preg_match('/(live|test)/', $apiKeysRow['path'], $matches);
+            $apiKey = $this->scopeConfig->getValue($apiKeysRow['path'], $apiKeysRow['scope'], $apiKeysRow['scope_id']);
```

Comment:
for lisibility and unicity, I will have move the code inside the forech to an other function

Reply:
done